benchmarks/conftest.py

```diff
@@ -137,14 +169,34 @@ def record(self, capsys: pytest.CaptureFixture[str], name, **parameters):
         elapsed = time() - start
         end_cpu = get_cpu_time_for_cgroup()
         elapsed_cpu = end_cpu - start_cpu
+
+        self.benchmarks[name] = Benchmark(
+            name=name,
+            wall=elapsed,
+            cpu=elapsed_cpu,
+            parameters=parameters,
+        )
         # FOR now we just print the outcome:
         parameters = " ".join(f"{k}={v}" for (k, v) in parameters.items())
         with capsys.disabled():
             print(
                 f"\nBENCHMARK RESULT: {name} {parameters} elapsed={elapsed:.3} (secs) CPU={elapsed_cpu:.3} (secs)\n"
             )
 
+    def output_results(self, filelike):
+        js = dict()
+        for b in self.benchmarks.values():
+            js[b.name] = b.to_json()
+        filelike.write(
+            json.dumps(js, indent=4).encode("utf8")
+        )
+
 
 @pytest.fixture(scope="session")
-def tahoe_benchmarker():
-    return Benchmarker()
+def tahoe_benchmarker(request):
+    bm = Benchmarker()
+    yield bm
+    fname = request.config.getoption("json_fname")
+    print('Writing benchmarks to "{fname}"')
+    with open(fname, "wb") as js_file:
+        bm.output_results(js_file)
```

Comment:
Can this be stated as a ticket?

I don't *necessarily* disagree but I don't really see how to do that -- and also it might be very confusing. 

That said, I guess it's kind of rude to "delete" (overwrite) files, so maybe a different solution would be to make sure they're unique (maybe even make the latest one a symlink?)

Reply:
For example, I routinely put `coverage erase` in scripts etc because otherwise it's easy to overstate coverage (due in part to the way it works, but ...)

Reply:
i guess the file could just be ... "a list of benchmark-results" instead -- is that what you meant?

Reply:
(did the last thing).